hr_attendance_calendar/models/hr_attendance_day.py

```diff
@@ -0,0 +1,297 @@
+
+    @api.multi
+    @api.depends('break_ids')
+    def _compute_break_max(self):
+        for att_day in self.filtered('break_ids'):
+            att_day.break_max = max(att_day.break_ids.mapped('duration'))
+
+    @api.multi
+    @api.depends('break_ids')
+    def _compute_break_total(self):
+        for att_day in self.filtered('break_ids'):
+            att_day.break_max = sum(att_day.break_ids.mapped('duration'))
+
+    @api.multi
+    @api.depends('worked_hours', 'due_hours')
+    def _compute_extra_hours(self):
+        for att_day in self.filtered('worked_hours'):
+            att_day.extra_hours = att_day.worked_hours - att_day.due_hours
+
+    @api.multi
+    def _compute_weighting(self):
+        return 1
+
+    ##########################################################################
+    #                             PUBLIC METHODS                             #
+    ##########################################################################
+    @api.model
+    def create(self, vals):
+        rd = super(HrDayAttendance, self).create(vals)
+
+        cal_att_date = fields.Date.from_string(rd.date)
+
+        # link to schedule (resource.calendar.attendance)
+        att_day = cal_att_date.weekday()
+        cal_att_ids = rd.employee_id.contract_id.working_hours.attendance_ids
+        current_cal_att = cal_att_ids.filtered(
+            lambda a: int(a.dayofweek) == att_day)
+
+        for cal_att_id in current_cal_att:
+            start = cal_att_id.date_from
+            start = start if start else date.min
+            end = cal_att_id.date_to
+            end = end if end else date.max
+            if start <= cal_att_date <= end:
+                rd.cal_att_ids += cal_att_id
+
+        # link to leaves (hr.holidays )
+        holiday_ids = self.env['hr.holidays'].search([
+            ('employee_id', '=', rd.employee_id.id),
+            ('type', '=', 'remove')])
+        for leave in holiday_ids:
+            start = fields.Date.from_string(leave.date_from)
+            end = fields.Date.from_string(leave.date_to)
+            if start <= cal_att_date <= end:
+                rd.leave_ids += leave
+
+        # find weighting hours
+        weighting = self.env['hr.attendance.weighting'].search([
+            ('day_of_week', '=', att_day),
+        ])
+        # todo: after settings remove
+        if not weighting:
+            rd.weighting = 1
+        else:
+            for category in weighting.category_ids:
+                if category in rd.category_ids:
+                    rd.weighting = weighting
+                    break
+
+        # find related attendance
+        rd.recompute_attendance()
+
+        # compute breaks
+        self.compute_breaks()
+
+        return rd
+
+    @api.multi
+    def compute_breaks(self):
+        for att_day in self:
+            att_day.break_ids.unlink()
+            if len(att_day.attendance_ids) <= 1:
+                # no break
+                continue
+            else:
```

Comment:
The else part is not needed because you use the `continue` before. You can just outdent all the following code.